test/e2e/apimachinery/resource_quota.go

```diff
@@ -1605,6 +1606,8 @@ var _ = SIGDescribe("ResourceQuota", func() {
 		usedResources[v1.ResourceLimitsMemory] = resource.MustParse("0")
 		err = waitForResourceQuota(ctx, f.ClientSet, f.Namespace.Name, resourceQuotaNotTerminating.Name, usedResources)
 		framework.ExpectNoError(err)
+		err = waitForResourceQuota(ctx, f.ClientSet, f.Namespace.Name, resourceQuotaTerminating.Name, usedResources)
+		framework.ExpectNoError(err)
```

Comment:
I've not been able to reproduce the flake locally, but this seems like it should fix the problem.

Reply:
There's another one:
```
[FAILED] pods "test-pod" is forbidden: exceeded quota: quota-terminating, requested: pods=1, used: pods=1, limited: pods=1
In [It] at: k8s.io/kubernetes/test/e2e/apimachinery/resource_quota.go:1573 @ 05/19/25 07:49:11.572
```

https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/131843/pull-kubernetes-e2e-kind/1924367954477060096

Reply:
But this one here also occurs:
```
[FAILED] pods "terminating-pod" is forbidden: exceeded quota: quota-terminating, requested: pods=1, used: pods=1, limited: pods=1
In [It] at: k8s.io/kubernetes/test/e2e/apimachinery/resource_quota.go:1611 @ 05/19/25 08:07:35.598
```

https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/131844/pull-kubernetes-e2e-kind/1924371184065974272

Reply:
Still trying to make sense of the test...

cc @carlory , you changed this in https://github.com/kubernetes/kubernetes/commit/eb0f003d25209c850b47a275358aea53252274b4#diff-e1a46008c4ddbb9f83858cb22ce56ca0f820b78142a1e8abbbb11aa59127a09b.

One observation: when it checks for "ignored the pod usage", it's not sufficient to check once and move on if it has the desired status. One has to check for a certain period of time to be certain that the kube-controller-manager doesn't change the status some time after the initial check.

Reply:
Another observation: using `ExpectNoError` with no additional explanation makes it hard to correlate failures with steps in the test because one first has to look up the source code. Even then line numbers are not always the same (think merging an older PR into a new master with concurrent changes).

Reply:
I added `ensureResourceQuota` before line 1573. If the ResourceQuota unexpectedly counts one terminating pod, then it'll show up more obviously than with just the quota error during admission of the pod update.

Reply:
/assign @carlory 

Do my changes make sense to you?

Reply:
Good catch, it is my mistake. I wanted to wait for the terminating quota to be updated, not the non-terminating one because we have checked it before deleting the pod1 when I wrote that PR.

This PR looks good to me.

/lgtm
/hold
feel free to unhold it.

Reply:
Please re-add LGTM, I missed one error check: https://github.com/kubernetes/kubernetes/compare/0f1fbce3a75aadc49fb6cae1c22941a336661066..3878f7e7489285a616ba354ff5912ce8abcc19e2

I think we can lift the hold. This PR only fixes one of the two flakes, but adds a bit better output for the other. We can always investigate the other separately.

Reply:
I've updated the description with a description of which flake gets fixed (fingers crossed!) and which doesn't. Both have the exact same failure message.

I'll refrain from annotating all `framework.ExpectError`, but this shows that test with repeated checks absolutely should use annotation.

Reply:
/hold cancel